### ISSUE TITLE
[Tables] deleteOne returns HTTP 500 and an internal server error if doing "$gt" (& co.) on partition key

### ISSUE THREAD

**Issue by hemidactylus** [stargate/data-api#1755]
Table:

```
CREATE TABLE default_keyspace.x (
    a int PRIMARY KEY
)
```

deleteOne command:

```
{
    "deleteOne": {
        "filter": {
            "a": {"$gt": 3}
        }
    }
}
```

Instead of a HTTP 200 and an error by the API, the response is a 500 and a direct lamentation straight from CQL:

```
{
    "errors": [
        {
            "message": "An unexpected server error occurred while processing the request. \n\nError Class: InvalidQueryException\nError Message: Only EQ and IN relation are supported on the partition key (unless you use the token() function) for DELETE statements",
            "scope": "",
            "errorCode": "UNEXPECTED_SERVER_ERROR",
            "id": "45c3de33-5efe-4fb0-89af-dc2f4e6d6ed9",
            "family": "SERVER",
            "title": "Unexpected server error"
        }
    ]
}
```

**Comment by amorton** [comment]
similar to #1758

retest to see if fixed by #1729

**Comment by Hazel-Datastax** [comment]
Confirmed that the issue has been fixed. Current error message is:

```json
{
    "errors": [
        {
            "message": "The Filtering using the non `$eq` filter operations can select more than one row, and so cannot be used with the `updateOne` and `deleteOne` commands as they can only modify one row.\n\nThe command used an invalid filter on the columns: a.\n\nResend the command only using `$eq` filters.",
            "scope": "FILTER",
            "errorCode": "UNSUPPORTED_FILTER_FOR_UPDATE_ONE_DELETE_ONE",
            "exceptionClass": "FilterException",
            "id": "70c00ed4-e868-44de-8fec-d535eb59a4c7",
            "family": "REQUEST",
            "title": "updateOne and deleteOne commands only support filtering using $eq"
        }
    ]
}
```
